sdk/communication/azure-communication-phonenumbers/src/main/java/com/azure/communication/phonenumbers/SipRoutingClientBuilder.java

```diff
@@ -0,0 +1,485 @@
+     */
+    public SipRoutingClient buildClient() {
+        validateRequiredFields();
+
+        if (this.version != null) {
+            logger.info("Build client for service version" + this.version.getVersion());
+        }
+
+        return createClientImpl(createAdminClientImpl());
+    }
+
+    /**
+     * Create asynchronous client applying CommunicationClientCredentialPolicy,
+     * UserAgentPolicy, RetryPolicy, and CookiePolicy.
+     * Additional HttpPolicies specified by additionalPolicies will be applied after them
+     *
+     * @return {@link SipRoutingAsyncClient} instance.
+     * @throws IllegalStateException If both {@link #retryOptions(RetryOptions)}
+     * and {@link #retryPolicy(RetryPolicy)} have been set.
+     */
+    public SipRoutingAsyncClient buildAsyncClient() {
+        validateRequiredFields();
+
+        if (this.version != null) {
+            logger.info("Build client for service version" + this.version.getVersion());
+        }
+
+        return createAsyncClientImpl(createAdminClientImpl());
+    }
+
+    SipRoutingClient createClientImpl(SipRoutingAdminClientImpl adminClientImpl) {
+        return new SipRoutingClient(adminClientImpl);
+    }
+
+    SipRoutingAsyncClient createAsyncClientImpl(SipRoutingAdminClientImpl adminClientImpl) {
+        return new SipRoutingAsyncClient(adminClientImpl);
+    }
+
+    HttpPipelinePolicy createAuthenticationPolicy() {
+        if (this.tokenCredential != null && this.azureKeyCredential != null) {
+            throw logger.logExceptionAsError(
+                new IllegalArgumentException("Both 'credential' and 'keyCredential' are set. Just one may be used."));
+        }
+        if (this.tokenCredential != null) {
+            return new BearerTokenAuthenticationPolicy(
+                this.tokenCredential, "https://communication.azure.com//.default");
+        } else if (this.azureKeyCredential != null) {
+            return new HmacAuthenticationPolicy(this.azureKeyCredential);
+        } else {
+            throw logger.logExceptionAsError(
+                new NullPointerException("Missing credential information while building a client."));
+        }
+    }
+
+    UserAgentPolicy createUserAgentPolicy(
+        String applicationId, String sdkName, String sdkVersion, Configuration configuration) {
+        return new UserAgentPolicy(applicationId, sdkName, sdkVersion, configuration);
+    }
+
+    HttpPipelinePolicy createRequestIdPolicy() {
+        return new RequestIdPolicy();
+    }
+
+    CookiePolicy createCookiePolicy() {
+        return new CookiePolicy();
+    }
+
+    HttpLoggingPolicy createHttpLoggingPolicy(HttpLogOptions httpLogOptions) {
+        return new HttpLoggingPolicy(httpLogOptions);
+    }
+
+    HttpLogOptions createDefaultHttpLogOptions() {
+        return new HttpLogOptions();
+    }
+
+    private void validateRequiredFields() {
+        Objects.requireNonNull(endpoint);
+    }
```

Comment:
Just a nit: since this are implementation you don't need to remove them, but the methods are so simple they could just be used inline.

Reply:
This is extracted into methods mainly for testing purposes. 
For now, I would keep it consistent with PhoneNumbersClientBuilder.